engine/src/main/java/org/terasology/rendering/dag/ConditionDependentNode.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import com.google.api.client.util.Lists;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * TODO: Add javadocs
+ */
+public abstract class ConditionDependentNode extends AbstractNode implements PropertyChangeListener {
```

Comment:
Module developers can already do what you are saying. Freedom is preserved! 😄 This class actually introduces an helper method `requiresCondition` which accepts a closure(lambda function). This function can be anything, a very good use case for this one is in `BlurPassesNode`.

```java
    private FBO sceneToneMapped;

    @Override
    public void initialise() {
        renderingConfig = config.getRendering();
        blur = worldRenderer.getMaterial("engine:prog.blur");

        renderingConfig.subscribe(RenderingConfig.BLUR_INTENSITY, this);
        requiresCondition(() -> renderingConfig.getBlurIntensity() != 0);
        requiresFBO(new FBOConfig(BLUR_0, HALF_SCALE, FBO.Type.DEFAULT), displayResolutionDependentFBOs);
        requiresFBO(new FBOConfig(BLUR_1, HALF_SCALE, FBO.Type.DEFAULT), displayResolutionDependentFBOs);
        requiresFBO(new FBOConfig(TONE_MAPPED, FULL_SCALE, FBO.Type.HDR), displayResolutionDependentFBOs);
    }
```

I was between the term `ConditionDependentNode` and `ConfigDependentNode`. `Condition` seemed more generic, as you mentioned. Another benefit of having this class is reducing huge amount of redundant code. (Same lines in many different nodes.) Therefore, for the time being, I would not remove this class. In long term goals, I would investigate better ways to connect current configurations with nodes.